### PR TITLE
chore(site): add InputGroup component

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Created a new component called InputGroup to make it easier for us to group input elements as we do in the search.

Related to: https://github.com/coder/coder/issues/13035

### CODE REVIEW

File: site/src/components/InputGroup/InputGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      {...props}
+      css={{
+        "&": {
+          display: "flex",
+          alignItems: "flex-start",
+        },
+
+        "& > *:hover": {
+          zIndex: 1,
+        },
```
Comment (Parkreiner) [inline site/src/components/InputGroup/InputGroup.tsx:15]: I'm not sure what the z Index is doing here. Is this to handle some MUI styling?

**Review by buenos-nachos — APPROVED** [review]
This looks straightforward to me. Just had a question some of the CSS

File: site/src/components/InputGroup/InputGroup.tsx (hunk 1 above)
Comment (Parkreiner) [reply site/src/components/InputGroup/InputGroup.tsx:15]: Oh, unless it's specifically to create a stacking context, but in that case, I feel like a comment would be good

File: site/src/components/InputGroup/InputGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        "& > *:not(:last-child)": {
+          marginRight: -1,
```
[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply site/src/components/InputGroup/InputGroup.tsx:18]: It is to void having two borders touching each other but I will left a comment to make it more clear.

**Comment by BrunoQuaresma** [comment]
@Parkreiner added a few comments explaining the styles. I hope this now makes more sense 🙏

**Pull request merged**